src/passes/Outlining.cpp

```diff
@@ -132,7 +132,15 @@ struct ReconstructStringifyWalker
                          : state == NotInSeq ? &existingBuilder
                                              : nullptr;
     if (builder) {
-      ASSERT_OK(builder->visit(curr));
+      if (auto* expr = curr->dynCast<Break>()) {
+        Type type = expr->value ? expr->value->type : Type::none;
+        ASSERT_OK(builder->visitBreakWithType(expr, type));
+      } else if (auto* expr = curr->dynCast<Switch>()) {
+        Type type = expr->value ? expr->value->type : Type::none;
+        ASSERT_OK(builder->visitSwitchWithType(expr, type));
+      } else {
+        ASSERT_OK(builder->visit(curr));
```

Comment:
It would be good to add an assertion somewhere to future-proof ourselves against the possibility that a new kind of `Expression` starts to need a `visit*WithType` variant. This could either be an `assert(!Properties::isBranch(curr))` here guarding the default `builder->visit(curr)` case, or it could be an assertion that `curr` is a `Break` or `Switch` inside `IRBuilder::getBranchValue`.

Reply:
done